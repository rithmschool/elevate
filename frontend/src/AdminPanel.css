.admin-panel{
  display: grid;
  grid-template-areas: "content side";
  grid-template-columns: 70% 30%;
  height: 100vh;
  grid-area: content;
  border: 1px solid black;
}

.admin-navbar {
  grid-area: side;
  border: 1px solid blue;
}

@media (max-width: 875px) {

}

@media only screen and (min-width:480px) and (max-width: 640px)  {
  .admin-panel{
    grid-template-columns: 100%;
  }

  .admin-navbar {
    display: none;
  }
}