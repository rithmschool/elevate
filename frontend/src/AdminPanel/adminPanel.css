.adminPanel_h1 {
  font-size: 2em;
}

.adminPanel_main {
  display: grid;
  grid-template-areas: "content side";
  grid-template-columns: 75% 25%;
  height: 100vh;
  background: #ECF0F5;
}

.adminPanel_panel {
  grid-area: content;
  font-size: 0.75rem;
  text-transform: capitalize;
  padding: 1.2em;
}

.adminPanel_navbar {
  grid-area: side;
  background: #222D32;
  color: lightgrey;
  padding: 0.7em;
}

@media only screen and (max-width: 640px)  {
  .adminPanel_main{
    grid-template-columns: 100%;
  }

  .adminPanel_navbar {
    display: none;
  }
}
