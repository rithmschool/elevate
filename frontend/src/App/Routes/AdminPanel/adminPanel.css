.adminPanel_h1 {
  font-size: 2em;
}

.adminPanel_main {
  display: grid;
  grid-template-areas: "content side";
  height: 100vh;
  background: #ECF0F5;
}

.adminPanel_main_showing {
  grid-template-columns: 75% 25%;
}

.adminPanel_main_docked {
  grid-template-columns: 93% 7%;
}

.adminPanel_navbar {
  position: relative; 
  overflow: hidden;
  background: #222D32;
  color: lightgrey;
}

.adminPanel_navbar_showing {
}

.adminPanel_navbar_docked {
  cursor: pointer;
  padding: 0;
}

.adminPanel_panel {
  font-size: 0.75rem;
  text-transform: capitalize;
  padding: 1.2em;
}

@media only screen and (max-width: 640px)  {
  .adminPanel_main_showing {
    grid-template-columns: 30% 70%;
  }
  .adminPanel_main_docked {
    grid-template-columns: 83% 17%;
  }
}

@media only screen and (min-width: 1200px)  {
  .adminPanel_main_showing {
    grid-template-columns: 80% 20%;
  }
  .adminPanel_main_docked {
    grid-template-columns: 96% 4%;
  }
}
