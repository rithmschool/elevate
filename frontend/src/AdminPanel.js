import React, { Component } from 'react';
import './AdminPanel.css'
import AdminNavBar from './AdminNavBar';
import AdminUserView from './AdminUserView';
import AdminTable from './AdminTable';
import ElevateApi from './ElevateApi';

const mql = window.matchMedia(`(max-width: 640px)`);

class AdminPanel extends Component {
  constructor(props){
    super(props);

    this.state = {
      view: '',
      sidebarDocked: mql.matches,
      sideBarOpen: false,
      users: null,
      questions:null,
      userDetail: null
    }
  }

  componentDidMount = async () => {
    mql.addListener(this.mediaQueryChanged);
    let users;
    let questions;

    try {
      users = await ElevateApi.getUsers();
      questions = await ElevateApi.getQuestions();
    } catch(err) {
      console.log(err)
      return err;
    }
    
    this.setState({ users, questions });
  }

  // get update users after delete a user in AdminUserView
  updateUserState = (users) => {
    this.setState({users})
  }

  changeView = (view) => {
    this.setState({ view })
  }

  mediaQueryChanged = () => {
    this.setState({ sidebarDocked: mql.matches });
  }

  toggleSidebar = () => {
    this.setState({ sideBarOpen: !this.state.sideBarOpen });
  }

  getUserDetail = async (userId) => {
    const user = await ElevateApi.getUser(userId);
    
    this.setState({ view: 'userDetail', userDetail: user });
  }

  render(){
    if (!this.state.users || !this.state.questions){
      return (<div>...Loading</div>);
    }

    return(
      <div className="admin-main">
        <div className="admin-panel">
          { mql.matches && <button onClick={this.toggleSidebar}>SIDEBAR</button> }
          <h1 className="admin-h1">Admin Panel</h1>
          { this.state.sideBarOpen && <AdminNavBar changeView={this.changeView} /> }
          { this.state.view === 'users' || this.state.view === 'questions' ? 
            <AdminTable tableObjs={ this.state[this.state.view] } 
                        getUserDetail={ this.getUserDetail }
                        view={ this.state.view } /> : null
          }
          {this.state.view === 'userDetail' && 
                                <AdminUserView 
                                  user={this.state.userDetail} 
                                  updateUserState={this.updateUserState} 
                                  changeView={this.changeView} /> 
          }
        </div>
        
        <div className="admin-navbar">
          <AdminNavBar changeView={this.changeView}/>
        </div>
      </div>
    )
  }
}

export default AdminPanel;
