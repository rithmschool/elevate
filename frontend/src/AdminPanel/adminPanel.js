import React from "react";
import "./adminPanel.css";

import AdminNavbar from "../AdminNavbar/adminNavbar";
import AdminUserView from "../AdminUserView/adminUserView";
import AdminTable from "../AdminTable/adminTable";
import ElevateApi from "../elevateApi";
import Spinner from "../Spinner/spinner";


const mql = window.matchMedia(`(max-width: 640px)`);

class AdminPanel extends React.Component {
  constructor(props){
    super(props);

    this.state = {
      view: "",
      sidebarDocked: mql.matches,
      sideBarOpen: false,
      users: null,
      questions:null,
      userDetail: null
    }
  }

  componentDidMount = async () => {
    mql.addListener(this.mediaQueryChanged);
    let users;
    let questions;

    try {
      users = await ElevateApi.getUsers();
      questions = await ElevateApi.getQuestions();
    } catch(err) {
      console.log(err)
      return err;
    }
    
    this.setState({ users, questions });
  }

  // get update users after delete a user in AdminUserView
  updateUserState = users => {
    this.setState({ users })
  }

  changeView = view => {
    this.setState({ view })
  }

  mediaQueryChanged = () => {
    this.setState({ sidebarDocked: mql.matches });
  }

  toggleSidebar = () => {
    this.setState(st => ({ sideBarOpen: !st.sideBarOpen }));
  }

  getUserDetail = async userId => {
    const user = await ElevateApi.getUser(userId);
    
    this.setState({ view: "userDetail", userDetail: user });
  }

  render(){
    if (!this.state.users || !this.state.questions){
      return <Spinner />;
    }

    return (
      <div className="adminPanel_main">

        <div className="adminPanel_panel">

          { mql.matches 
              && <button onClick={this.toggleSidebar}>SIDEBAR</button> }

          <h1 className="adminPanel_h1">Admin Panel</h1>

          { this.state.sideBarOpen 
              && <AdminNavbar changeView={this.changeView} /> }

          { ["users", "questions"].includes(this.state.view)
              && <AdminTable 
                   tableObjs={ this.state[this.state.view] } 
                   getUserDetail={ this.getUserDetail }
                   view={ this.state.view } /> }

          { this.state.view === "userDetail" 
              && <AdminUserView 
                   user={ this.state.userDetail }
                   updateUserState={ this.updateUserState }
                   changeView={ this.changeView } /> }
        </div>
        
        <div className="adminPanel_navbar">
          <AdminNavbar changeView={ this.changeView } />
        </div>

      </div>
    )
  }
}

export default AdminPanel;
