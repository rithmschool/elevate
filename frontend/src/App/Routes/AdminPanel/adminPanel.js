import React from "react";
import "./adminPanel.css";
import { Switch } from "react-router-dom";

import AdminPrivateRoute from "../adminPrivateRoute";
import AdminNavbar from "./AdminNavbar/adminNavbar";
import AdminUserView from "./AdminUserView/adminUserView";
import AdminDocView from "./AdminDocView/adminDocView";
import AdminTable from "./AdminTable/adminTable";
import ElevateApi from "../../../elevateApi";
import Spinner from "../../Spinner/spinner";
import PanelToggleBtn from "./PanelToggleBtn/panelToggleBtn";

class AdminPanel extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      sideBarOpen: true,
      users: null,
      questions: null,
      documents: null
    };
  }

  componentDidMount = async () => {
    let users;
    let questions;
    let documents;

    try {
      users = await ElevateApi.getUsers();
      questions = await ElevateApi.getQuestions();
      documents = await ElevateApi.getDocuments();
    } catch (err) {
      console.log(err);
      return err;
    }

    this.setState({ users, questions, documents });
  };

  // get update users after delete a user in AdminUserView
  updateUserState = users => {
    this.setState({ users });
  };

  toggleSidebar = () => {
    this.setState(st => ({ sideBarOpen: !st.sideBarOpen }));
  };

  render() {
    if (!this.state.users || !this.state.questions) {
      return <Spinner />;
    }

    const position = this.state.sideBarOpen ? "showing" : "docked";
    return (
      <div className={`adminPanel_main adminPanel_main_${position}`}>
        <div className="adminPanel_panel">
          <h1 className="adminPanel_h1">Admin Panel</h1>

          <Switch>
            <AdminPrivateRoute
              exact
              path="/dashboard/admin/users"
              render={props => (
                <AdminTable
                  tableType={"users"}
                  tableObjs={this.state.users}
                  {...props}
                />
              )}
            />
            <AdminPrivateRoute
              exact
              path="/dashboard/admin/questions"
              render={props => (
                <AdminTable
                  tableType={"questions"}
                  tableObjs={this.state.questions}
                  {...props}
                />
              )}
            />
            <AdminPrivateRoute
              exact
              path="/dashboard/admin/documents"
              render={props => (
                <AdminTable
                  tableType={"documents"}
                  tableObjs={this.state.documents}
                  {...props}
                />
              )}
            />
            <AdminPrivateRoute
              exact
              path="/dashboard/admin/users/:userId"
              render={props => (
                <AdminUserView
                  updateUserState={this.updateUserState}
                  {...props}
                />
              )}
            />
            <AdminPrivateRoute
              exact
              path="/dashboard/admin/documents/:docId"
              render={props => (
                <AdminDocView allDocs={this.state.documents} {...props} />
              )}
            />
          </Switch>
        </div>

        <div className={`adminPanel_navbar adminPanel_navbar_${position}`}>
          <div>
            <PanelToggleBtn
              toggleSidebar={this.toggleSidebar}
              direction={this.state.sideBarOpen ? "toggleRight" : "toggleLeft"}
            />
          </div>

          <AdminNavbar
            position={this.state.sideBarOpen}
            toggleSidebar={this.toggleSidebar}
          />
        </div>
      </div>
    );
  }
}

export default AdminPanel;
