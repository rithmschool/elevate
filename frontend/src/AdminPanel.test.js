import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from "enzyme-to-json";
import AdminPanel from './AdminPanel';

describe('AdminPanel', function() {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<AdminPanel />);
  });

  it('renders without crashing', function () {
    shallow(<AdminPanel />);
  });
  
  it('matches snapshot', function () {
    const serialized = toJson(wrapper);

    expect(serialized).toMatchSnapshot();
  });

  it('has states', function () {
    expect(wrapper.state('view')).toEqual('');
    expect(wrapper.state('sidebarDocked')).toEqual(false);
    expect(wrapper.state('sideBarOpen')).toEqual(false);
  });

  it('has div with admin-main class', function () {
    expect(wrapper.find('div.admin-main')).toHaveLength(1);
  });

  it('has div with admin-panel class', function () {
    expect(wrapper.find('div.admin-main')).toHaveLength(1);
  });

  it('changes view state on click', function () {
    wrapper.find('div[id="users"]').simulate('click');
    expect(wrapper.state('view')).toEqual('users');

    wrapper.find('div[id="invoices"]').simulate('click');
    expect(wrapper.state('view')).toEqual('invoices');

    wrapper.find('div[id="templates"]').simulate('click');
    expect(wrapper.state('view')).toEqual('templates');

    wrapper.find('div[id="calendar"]').simulate('click');
    expect(wrapper.state('view')).toEqual('calendar');
  });

  it('renders the users table when view state is users', function () {
    wrapper.setState({users: []})
    wrapper.find('div[id="users"]').simulate('click');
    wrapper.update();
    
    expect(wrapper.find('table[id="users-table"]')).toHaveLength(1);
  });

  it('show expect user data in the table', function () {
    wrapper.setState({
      users: [{
        id: 17, 
        email: "testadmin@test.com", 
        is_admin: true, 
        first_name: "admin", 
        last_name: "test", 
        current_company:"testcompany", 
        hire_date: "2018-06-23", 
        needs:"To test user data", 
        goals:"Test pass"
      }]
    })
    wrapper.find('div[id="users"]').simulate('click')
    wrapper.update();

    const rows = wrapper.find('table[id="users-table"]')
    expect(rows.length).toEqual(1);

    const dataRow = rows.first().find('td').map(column => column.text())
    expect(dataRow.length).toEqual(6)
    expect(dataRow[0]).toEqual("17")
    expect(dataRow[1]).toEqual("admin test")
    expect(dataRow[2]).toEqual("testcompany")
    expect(dataRow[3]).toEqual("2018-06-23")
    expect(dataRow[4]).toEqual("To test user data")
    expect(dataRow[5]).toEqual("Test pass")
  });
});