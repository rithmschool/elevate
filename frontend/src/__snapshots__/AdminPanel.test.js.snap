// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPanel matches snapshot 1`] = `
<AdminPanel>
  <div>
    ...Loading
  </div>
</AdminPanel>
`;
